Hmmm. I'm not sure why `TestCSVImportCanBeResumed` is failing under `testrace` (locally and in CI). [CI claims ](https://teamcity.cockroachdb.com/viewLog.html?buildId=4179807&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests&logTab=)the test didn't fail on my previous commit. But now, when I check out the previous [commit](https://github.com/cockroachdb/cockroach/commit/dbdc19252df4de0aec8d187e1951a7d1a5bd7d39) or even on master, and rerun the test locally, it fails consistently with the following stacktrace. 
```
e5
github.com/cockroachdb/cockroach/pkg/util/log.init()
        /Users/michaelbutler/go/src/github.com/cockroachdb/cockroach/pkg/util/log/file_names.go:71 +0x1ca fp=0xc00035f968 sp=0xc00035f6f8 pc=0x4d74bea
runtime.doInit(0x106bf440)
        /usr/local/opt/go/libexec/src/runtime/proc.go:6498 +0x123 fp=0xc00035faa0 sp=0xc00035f968 pc=0x404af03
runtime.doInit(0x106b49a0)
        /usr/local/opt/go/libexec/src/runtime/proc.go:6475 +0x71 fp=0xc00035fbd8 sp=0xc00035faa0 pc=0x404ae51
runtime.doInit(0x106ad9a0)
        /usr/local/opt/go/libexec/src/runtime/proc.go:6475 +0x71 fp=0xc00035fd10 sp=0xc00035fbd8 pc=0x404ae51
runtime.doInit(0x106dbba0)
        /usr/local/opt/go/libexec/src/runtime/proc.go:6475 +0x71 fp=0xc00035fe48 sp=0xc00035fd10 pc=0x404ae51
runtime.doInit(0x10695880)
        /usr/local/opt/go/libexec/src/runtime/proc.go:6475 +0x71 fp=0xc00035ff80 sp=0xc00035fe48 pc=0x404ae51
runtime.main()
        /usr/local/opt/go/libexec/src/runtime/proc.go:238 +0x1e6 fp=0xc00035ffe0 sp=0xc00035ff80 pc=0x403d826
runtime.goexit()
        /usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc00035ffe8 sp=0xc00035ffe0 pc=0x4070e01

rax    0x0
rbx    0x1a3bb600
rcx    0x7ff7bfefed68
rdx    0x0
rdi    0x103
rsi    0x6
rbp    0x7ff7bfefed90
rsp    0x7ff7bfefed68
r8     0x0
r9     0x0
r10    0x0
r11    0x246
r12    0x103
r13    0x50
r14    0x6
r15    0x16
rip    0x7ff800a87112
rflags 0x246
cs     0x7
fs     0x0
gs     0x0
FAIL    github.com/cockroachdb/cockroach/pkg/ccl/importccl      2.587s
FAIL
make: *** [Makefile:1081: testrace] Error 1
```

--- a/pkg/ccl/importccl/exportcsv.go
+++ b/pkg/ccl/importccl/exportcsv.go
@@ -16,6 +16,7 @@ import (
 	"strings"
 
 	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
@@ -84,7 +85,7 @@ func (c *csvExporter) Len() int {
 	return c.buf.Len()
 }
 
-func (c *csvExporter) FileName(spec execinfrapb.CSVWriterSpec, part string) string {
+func (c *csvExporter) FileName(spec execinfrapb.ExportSpec, part string) string {
 	pattern := exportFilePatternDefault
 	if spec.NamePattern != "" {
 		pattern = spec.NamePattern
@@ -98,11 +99,11 @@ func (c *csvExporter) FileName(spec execinfrapb.CSVWriterSpec, part string) stri
 	return fileName
 }
 
-func newCSVExporter(sp execinfrapb.CSVWriterSpec) *csvExporter {
+func newCSVExporter(sp execinfrapb.ExportSpec) *csvExporter {
 	buf := bytes.NewBuffer([]byte{})
 	var exporter *csvExporter
-	switch sp.CompressionCodec {
-	case execinfrapb.FileCompression_Gzip:
+	switch sp.Format.Compression {
+	case roachpb.IOFileFormat_Gzip:
 		{
 			writer := gzip.NewWriter(buf)
 			exporter = &csvExporter{
@@ -119,8 +120,8 @@ func newCSVExporter(sp execinfrapb.CSVWriterSpec) *csvExporter {
 			}
 		}
 	}
-	if sp.Options.Comma != 0 {
-		exporter.csvWriter.Comma = sp.Options.Comma
+	if sp.Format.Csv.Comma != 0 {
+		exporter.csvWriter.Comma = sp.Format.Csv.Comma
 	}
 	return exporter
 }
@@ -128,7 +129,7 @@ func newCSVExporter(sp execinfrapb.CSVWriterSpec) *csvExporter {
 func newCSVWriterProcessor(
 	flowCtx *execinfra.FlowCtx,
 	processorID int32,
-	spec execinfrapb.CSVWriterSpec,
+	spec execinfrapb.ExportSpec,
 	input execinfra.RowSource,
 	output execinfra.RowReceiver,
 ) (execinfra.Processor, error) {
@@ -149,7 +150,7 @@ func newCSVWriterProcessor(
 type csvWriter struct {
 	flowCtx     *execinfra.FlowCtx
 	processorID int32
-	spec        execinfrapb.CSVWriterSpec
+	spec        execinfrapb.ExportSpec
 	input       execinfra.RowSource
 	out         execinfra.ProcOutputHelper
 	output      execinfra.RowReceiver
@@ -186,8 +187,8 @@ func (sp *csvWriter) Run(ctx context.Context) {
 		writer := newCSVExporter(sp.spec)
 
 		var nullsAs string
-		if sp.spec.Options.NullEncoding != nil {
-			nullsAs = *sp.spec.Options.NullEncoding
+		if sp.spec.Format.Csv.NullEncoding != nil {
+			nullsAs = *sp.spec.Format.Csv.NullEncoding
 		}
 		f := tree.NewFmtCtx(tree.FmtExport)
 		defer f.Close()
@@ -220,7 +221,7 @@ func (sp *csvWriter) Run(ctx context.Context) {
 
 				for i, ed := range row {
 					if ed.IsNull() {
-						if sp.spec.Options.NullEncoding != nil {
+						if sp.spec.Format.Csv.NullEncoding != nil {
 							csvRow[i] = nullsAs
 							continue
 						} else {

--- a/pkg/ccl/importccl/exportcsv_test.go
+++ b/pkg/ccl/importccl/exportcsv_test.go
@@ -276,16 +276,19 @@ func TestExportOrder(t *testing.T) {
 }
 
 // parquetTest provides information to validate a test of EXPORT PARQUET. All
-// fields below name and stmt validate some aspect of the exported parquet file.
+// fields below the stmt validate some aspect of the exported parquet file.
 // If a validation field is empty, then that field will not be used in the test.
 type parquetTest struct {
-	// name is the name of the test.
-	name string
+	// filePrefix provides the parquet file name in front of the parquetExportFilePattern
+	filePrefix string
+
+	// fileSuffix provides the compression type, if any, of the parquet file
+	fileSuffix string
 
 	// prep contains sql commands that will execute before the stmt.
 	prep []string
 
-	// stmt contains the EXPORT PARQUET statement.
+	// stmt contains the EXPORT PARQUET sql statement to test
 	stmt string
 
 	// colNames provides the expected column names for the parquet file.
@@ -427,7 +430,7 @@ false),(3, 'Carl', 1, 34.214,true),(4, 'Alex', 3, 14.3, NULL), (5, 'Bobby', 2, 3
 
 	tests := []parquetTest{
 		{
-			name: "basic",
+			filePrefix: "basic",
 			stmt: `EXPORT INTO PARQUET 'nodelocal://0/basic' FROM SELECT *
 							FROM foo WHERE y IS NOT NULL ORDER BY y ASC LIMIT 2 `,
 			colNames: []string{"i", "x", "y", "z", "a"},
@@ -435,7 +438,7 @@ false),(3, 'Carl', 1, 34.214,true),(4, 'Alex', 3, 14.3, NULL), (5, 'Bobby', 2, 3
 				{int64(2), "Bob", int64(2), 24.1, false}},
 		},
 		{
-			name: "null_vals",
+			filePrefix: "null_vals",
 			stmt: `EXPORT INTO PARQUET 'nodelocal://0/null_vals' FROM SELECT *
 							FROM foo ORDER BY x ASC LIMIT 2`,
 			vals: [][]interface{}{
@@ -443,13 +446,13 @@ false),(3, 'Carl', 1, 34.214,true),(4, 'Alex', 3, 14.3, NULL), (5, 'Bobby', 2, 3
 				{int64(4), "Alex", int64(3), 14.3, nil}},
 		},
 		{
-			name: "colname",
+			filePrefix: "colname",
 			stmt: `EXPORT INTO PARQUET 'nodelocal://0/colname' FROM SELECT avg(z), min(y) AS baz
 							FROM foo`,
 			colNames: []string{"avg", "baz"},
 		},
 		{
-			name: "nullable",
+			filePrefix: "nullable",
 			stmt: `EXPORT INTO PARQUET 'nodelocal://0/nullable' FROM SELECT y,z,x
 							FROM foo`,
 			colFieldRepType: []parquet.FieldRepetitionType{
@@ -465,7 +468,7 @@ false),(3, 'Carl', 1, 34.214,true),(4, 'Alex', 3, 14.3, NULL), (5, 'Bobby', 2, 3
 			// I already verified that the vendor's parquet writer can write arrays
 			// with null values just fine, so EXPORT PARQUET is bug free; however this
 			// roundtrip test would fail.
-			name: "arrays",
+			filePrefix: "arrays",
 			prep: []string{"CREATE TABLE atable (i INT PRIMARY KEY, x INT[])",
 				"INSERT INTO atable VALUES (1, ARRAY[1,2]), (2, ARRAY[2]), (3,ARRAY[1,13,5]),(4, NULL),(5, ARRAY[])"},
 			stmt:     `EXPORT INTO PARQUET 'nodelocal://0/arrays' FROM SELECT * FROM atable`,
@@ -478,25 +481,41 @@ false),(3, 'Carl', 1, 34.214,true),(4, 'Alex', 3, 14.3, NULL), (5, 'Bobby', 2, 3
 				{int64(5), []interface{}{}},
 			},
 		},
+		{
+			filePrefix: "compress_gzip",
+			fileSuffix: ".gz",
+			stmt: `EXPORT INTO PARQUET 'nodelocal://0/compress_gzip' WITH compression = gzip
+							FROM SELECT * FROM foo`,
+		},
+		{
+			filePrefix: "compress_snappy",
+			fileSuffix: ".snappy",
+			stmt: `EXPORT INTO PARQUET 'nodelocal://0/compress_snappy' WITH compression = snappy
+							FROM SELECT * FROM foo `,
+		},
+		{
+			filePrefix: "uncompress",
+			stmt: `EXPORT INTO PARQUET 'nodelocal://0/uncompress'
+							FROM SELECT * FROM foo `,
+		},
 	}
 
 	for _, test := range tests {
-		t.Logf("Test %s", test.name)
+		t.Logf("Test %s", test.filePrefix)
 		if test.prep != nil {
 			for _, cmd := range test.prep {
 				sqlDB.Exec(t, cmd)
 			}
 		}
 		sqlDB.Exec(t, test.stmt)
-		paths, err := filepath.Glob(filepath.Join(dir, test.name,
-			parquetExportFilePattern))
+
+		paths, err := filepath.Glob(filepath.Join(dir, test.filePrefix, parquetExportFilePattern+test.fileSuffix))
 		require.NoError(t, err)
 
 		require.Equal(t, 1, len(paths))
 
 		err = validateParquetFile(t, paths[0], test)
 		require.NoError(t, err)
-
 	}
 }
 

--- a/pkg/ccl/importccl/exportparquet.go
+++ b/pkg/ccl/importccl/exportparquet.go
@@ -15,6 +15,7 @@ import (
 	"strings"
 
 	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
@@ -40,6 +41,7 @@ type parquetExporter struct {
 	parquetWriter  *goparquet.FileWriter
 	schema         *parquetschema.SchemaDefinition
 	parquetColumns []parquetColumn
+	compression    roachpb.IOFileFormat_Compression
 }
 
 // Write appends a record to a parquet file.
@@ -65,7 +67,7 @@ func (c *parquetExporter) Bytes() []byte {
 
 func (c *parquetExporter) ResetBuffer() {
 	c.buf.Reset()
-	c.parquetWriter = buildFileWriter(c.buf, c.schema)
+	c.parquetWriter = c.buildFileWriter()
 }
 
 // Len returns length of the buffer with content.
@@ -73,22 +75,44 @@ func (c *parquetExporter) Len() int {
 	return c.buf.Len()
 }
 
-func (c *parquetExporter) FileName(spec execinfrapb.ParquetWriterSpec, part string) string {
+func (c *parquetExporter) FileName(spec execinfrapb.ExportSpec, part string) string {
 	pattern := exportParquetFilePatternDefault
 	if spec.NamePattern != "" {
 		pattern = spec.NamePattern
 	}
 
 	fileName := strings.Replace(pattern, exportFilePatternPart, part, -1)
-
+	suffix := ""
+	switch spec.Format.Compression {
+	case roachpb.IOFileFormat_Gzip:
+		suffix = ".gz"
+	case roachpb.IOFileFormat_Snappy:
+		suffix = ".snappy"
+	}
+	fileName += suffix
 	return fileName
+}
+
+func (c *parquetExporter) buildFileWriter() *goparquet.FileWriter {
+	var parquetCompression parquet.CompressionCodec
+	switch c.compression {
+	case roachpb.IOFileFormat_Gzip:
+		parquetCompression = parquet.CompressionCodec_GZIP
+	case roachpb.IOFileFormat_Snappy:
+		parquetCompression = parquet.CompressionCodec_SNAPPY
+	default:
+		parquetCompression = parquet.CompressionCodec_UNCOMPRESSED
+	}
+	pw := goparquet.NewFileWriter(c.buf,
+		goparquet.WithCompressionCodec(parquetCompression),
+		goparquet.WithSchemaDefinition(c.schema),
+	)
+	return pw
 }
 
 // newParquetExporter creates a new parquet file writer, defines the parquet
 // file schema, and initializes a new parquetExporter.
-func newParquetExporter(
-	sp execinfrapb.ParquetWriterSpec, typs []*types.T,
-) (*parquetExporter, error) {
+func newParquetExporter(sp execinfrapb.ExportSpec, typs []*types.T) (*parquetExporter, error) {
 
 	var exporter *parquetExporter
 
@@ -103,6 +127,7 @@ func newParquetExporter(
 		buf:            buf,
 		schema:         schema,
 		parquetColumns: parquetColumns,
+		compression:    sp.Format.Compression,
 	}
 	return exporter, nil
 }
@@ -120,10 +145,10 @@ type parquetColumn struct {
 }
 
 // newParquetColumns creates a list of parquet columns, given the input relation's column types.
-func newParquetColumns(typs []*types.T, sp execinfrapb.ParquetWriterSpec) ([]parquetColumn, error) {
+func newParquetColumns(typs []*types.T, sp execinfrapb.ExportSpec) ([]parquetColumn, error) {
 	parquetColumns := make([]parquetColumn, len(typs))
 	for i := 0; i < len(typs); i++ {
-		parquetCol, err := newParquetColumn(typs[i], sp.ColNames[i], sp.ColNullability[i])
+		parquetCol, err := newParquetColumn(typs[i], sp.ColNames[i], sp.Format.Parquet.ColNullability[i])
 		if err != nil {
 			return nil, err
 		}
@@ -289,21 +314,10 @@ func newParquetSchema(parquetFields []parquetColumn) *parquetschema.SchemaDefini
 	return schemaDefinition
 }
 
-func buildFileWriter(
-	buf *bytes.Buffer, schema *parquetschema.SchemaDefinition,
-) *goparquet.FileWriter {
-	pw := goparquet.NewFileWriter(buf,
-		// TODO(MB): allow for user defined compression
-		goparquet.WithCompressionCodec(parquet.CompressionCodec_SNAPPY),
-		goparquet.WithSchemaDefinition(schema),
-	)
-	return pw
-}
-
 func newParquetWriterProcessor(
 	flowCtx *execinfra.FlowCtx,
 	processorID int32,
-	spec execinfrapb.ParquetWriterSpec,
+	spec execinfrapb.ExportSpec,
 	input execinfra.RowSource,
 	output execinfra.RowReceiver,
 ) (execinfra.Processor, error) {
@@ -324,7 +338,7 @@ func newParquetWriterProcessor(
 type parquetWriterProcessor struct {
 	flowCtx     *execinfra.FlowCtx
 	processorID int32
-	spec        execinfrapb.ParquetWriterSpec
+	spec        execinfrapb.ExportSpec
 	input       execinfra.RowSource
 	out         execinfra.ProcOutputHelper
 	output      execinfra.RowReceiver

--- a/pkg/roachpb/io-formats.proto
+++ b/pkg/roachpb/io-formats.proto
@@ -24,6 +24,7 @@ message IOFileFormat {
     PgCopy = 4;
     PgDump = 5;
     Avro = 6;
+    Parquet = 7;
   }
 
   optional FileFormat format = 1 [(gogoproto.nullable) = false];
@@ -33,12 +34,14 @@ message IOFileFormat {
   optional MysqldumpOptions mysql_dump = 9 [(gogoproto.nullable) = false];
   optional PgDumpOptions pg_dump = 6 [(gogoproto.nullable) = false];
   optional AvroOptions avro = 8 [(gogoproto.nullable) = false];
+  optional ParquetOptions parquet = 10 [(gogoproto.nullable) = false];
 
   enum Compression {
     Auto = 0;
     None = 1;
     Gzip = 2;
     Bzip = 3;
+    Snappy = 4;
   }
   optional Compression compression = 5 [(gogoproto.nullable) = false];
   // If true, don't abort on failures but instead save the offending row and keep on.
@@ -62,9 +65,6 @@ message CSVOptions {
   // Indicates the number of rows to import per CSV file.
   // Must be a non-zero positive number. 
   optional int64 row_limit = 6 [(gogoproto.nullable) = false];
-}
-
-message ParquetOptions {
 }
 
 // MySQLOutfileOptions describe the format of mysql's outfile.
@@ -152,4 +152,9 @@ message AvroOptions {
   optional int32 max_record_size = 4 [(gogoproto.nullable) = false];
   optional int32 record_separator = 5 [(gogoproto.nullable) = false];
   optional int64 row_limit = 6 [(gogoproto.nullable) = false];
+}
+
+message ParquetOptions {
+  // col_nullability specifies which columns allow null values in the exported parquet file.
+  repeated bool col_nullability = 1 ;
 }

--- a/pkg/sql/colexec/colbuilder/execplan.go
+++ b/pkg/sql/colexec/colbuilder/execplan.go
@@ -244,8 +244,7 @@ var (
 	errReadImportWrap                 = errors.New("core.ReadImport is not supported")
 	errBackupDataWrap                 = errors.New("core.BackupData is not supported")
 	errBackfillerWrap                 = errors.New("core.Backfiller is not supported (not an execinfra.RowSource)")
-	errCSVWriterWrap                  = errors.New("core.CSVWriter is not supported (not an execinfra.RowSource)")
-	errParquetWriterWrap              = errors.New("core.ParquetWriter is not supported (not an execinfra.RowSource)")
+	errExporterWrap                   = errors.New("core.Exporter is not supported (not an execinfra.RowSource)")
 	errSamplerWrap                    = errors.New("core.Sampler is not supported (not an execinfra.RowSource)")
 	errSampleAggregatorWrap           = errors.New("core.SampleAggregator is not supported (not an execinfra.RowSource)")
 	errExperimentalWrappingProhibited = errors.New("wrapping for non-JoinReader and non-LocalPlanNode cores is prohibited in vectorize=experimental_always")
@@ -271,10 +270,8 @@ func canWrap(mode sessiondatapb.VectorizeExecMode, spec *execinfrapb.ProcessorSp
 		return errBackfillerWrap
 	case spec.Core.ReadImport != nil:
 		return errReadImportWrap
-	case spec.Core.CSVWriter != nil:
-		return errCSVWriterWrap
-	case spec.Core.ParquetWriter != nil:
-		return errParquetWriterWrap
+	case spec.Core.Exporter != nil:
+		return errExporterWrap
 	case spec.Core.Sampler != nil:
 		return errSamplerWrap
 	case spec.Core.SampleAggregator != nil:

--- a/pkg/sql/distsql_physical_planner.go
+++ b/pkg/sql/distsql_physical_planner.go
@@ -3845,31 +3845,14 @@ func (dsp *DistSQLPlanner) createPlanForExport(
 	}
 
 	var core execinfrapb.ProcessorCoreUnion
-
-	if n.csvOpts != nil {
-		core.CSVWriter = &execinfrapb.CSVWriterSpec{
-			Destination:      n.destination,
-			NamePattern:      n.fileNamePattern,
-			Options:          *n.csvOpts,
-			ChunkRows:        int64(n.chunkRows),
-			ChunkSize:        n.chunkSize,
-			CompressionCodec: n.fileCompression,
-			UserProto:        planCtx.planner.User().EncodeProto(),
-		}
-	} else if n.parquetOpts != nil {
-		core.ParquetWriter = &execinfrapb.ParquetWriterSpec{
-			Destination:    n.destination,
-			NamePattern:    n.fileNamePattern,
-			Options:        *n.parquetOpts,
-			ChunkRows:      int64(n.chunkRows),
-			ChunkSize:      n.chunkSize,
-			UserProto:      planCtx.planner.User().EncodeProto(),
-			ColNames:       n.colNames,
-			ColNullability: n.colNullability,
-		}
-	} else {
-		return nil, errors.AssertionFailedf("parquetOpts and csvOpts are both empty. " +
-			"One must be not nil")
+	core.Exporter = &execinfrapb.ExportSpec{
+		Destination: n.destination,
+		NamePattern: n.fileNamePattern,
+		Format:      n.format,
+		ChunkRows:   int64(n.chunkRows),
+		ChunkSize:   n.chunkSize,
+		ColNames:    n.colNames,
+		UserProto:   planCtx.planner.User().EncodeProto(),
 	}
 
 	resTypes := make([]*types.T, len(colinfo.ExportColumns))

--- a/pkg/sql/execinfra/version.go
+++ b/pkg/sql/execinfra/version.go
@@ -39,17 +39,20 @@ import "github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 //
 // ATTENTION: When updating these fields, add a brief description of what
 // changed to the version history below.
-const Version execinfrapb.DistSQLVersion = 59
+const Version execinfrapb.DistSQLVersion = 60
 
 // MinAcceptedVersion is the oldest version that the server is compatible with.
 // A server will not accept flows with older versions.
-const MinAcceptedVersion execinfrapb.DistSQLVersion = 58
+const MinAcceptedVersion execinfrapb.DistSQLVersion = 60
 
 /*
 
 **  VERSION HISTORY **
 
 Please add new entries at the top.
+
+- Version: 60 (MinAcceptedVersion: 60):
+ - Deprecated ExportWriterSpec and ParquetWriterSpec and merged them into ExportSpec
 
 - Version: 59 (MinAcceptedVersion: 58)
   - final_regr_sxx, final_regr_sxy, and final_regr_syy aggregate functions were

--- a/pkg/sql/execinfrapb/api.go
+++ b/pkg/sql/execinfrapb/api.go
@@ -68,12 +68,7 @@ func (m *BackupDataSpec) User() security.SQLUsername {
 }
 
 // User accesses the user field.
-func (m *CSVWriterSpec) User() security.SQLUsername {
-	return m.UserProto.Decode()
-}
-
-// User accesses the user field.
-func (m *ParquetWriterSpec) User() security.SQLUsername {
+func (m *ExportSpec) User() security.SQLUsername {
 	return m.UserProto.Decode()
 }
 

--- a/pkg/sql/execinfrapb/flow_diagram.go
+++ b/pkg/sql/execinfrapb/flow_diagram.go
@@ -520,8 +520,8 @@ func (c *ReadImportDataSpec) summary() (string, []string) {
 }
 
 // summary implements the diagramCellType interface.
-func (s *CSVWriterSpec) summary() (string, []string) {
-	return "CSVWriter", []string{s.Destination}
+func (s *ExportSpec) summary() (string, []string) {
+	return "Exporter", []string{s.Destination}
 }
 
 // summary implements the diagramCellType interface.

--- a/pkg/sql/execinfrapb/processors.proto
+++ b/pkg/sql/execinfrapb/processors.proto
@@ -101,7 +101,7 @@ message ProcessorCoreUnion {
   optional BackfillerSpec backfiller = 11;
   optional ReadImportDataSpec readImport = 13;
   reserved 14;
-  optional CSVWriterSpec CSVWriter = 20;
+  reserved 20;
   optional SamplerSpec Sampler = 15;
   optional SampleAggregatorSpec SampleAggregator = 16;
   reserved 17;
@@ -123,8 +123,7 @@ message ProcessorCoreUnion {
   optional FiltererSpec filterer = 34;
   optional StreamIngestionDataSpec streamIngestionData = 35;
   optional StreamIngestionFrontierSpec streamIngestionFrontier = 36;
-  optional ParquetWriterSpec ParquetWriter = 37;
-
+  optional ExportSpec exporter = 37;
   reserved 6, 12;
 }
 

--- a/pkg/sql/execinfrapb/processors_bulk_io.proto
+++ b/pkg/sql/execinfrapb/processors_bulk_io.proto
@@ -265,62 +265,27 @@ message SplitAndScatterSpec {
   repeated TenantRekey tenant_rekeys = 3 [(gogoproto.nullable) = false];
 }
 
-// FileCompression list of the compression codecs which are currently
-// supported for CSVWriter spec
-enum FileCompression {
-  None = 0;
-  Gzip = 1;
-}
-
-// CSVWriterSpec is the specification for a processor that consumes rows and
-// writes them to CSV files at uri. It outputs a row per file written with
+// ExporterSpec is the specification for a processor that consumes rows and
+// writes them to Parquet or CSV files at uri. It outputs a row per file written with
 // the file name, row count and byte size.
-message CSVWriterSpec {
+message ExportSpec {
   // destination as a cloud.ExternalStorage URI pointing to an export store
   // location (directory).
   optional string destination = 1 [(gogoproto.nullable) = false];
   optional string name_pattern = 2 [(gogoproto.nullable) = false];
-  optional roachpb.CSVOptions options = 3 [(gogoproto.nullable) = false];
-  // chunk_rows is num rows to write per file. 0 = no limit.
-  optional int64 chunk_rows = 4 [(gogoproto.nullable) = false];
-  // chunk_size is the target byte size per file.
-  optional int64 chunk_size = 7 [(gogoproto.nullable) = false];
-
-  // compression_codec specifies compression used for exported file.
-  optional FileCompression compression_codec = 5 [(gogoproto.nullable) = false];
-
-  // User who initiated the export. This is used to check access privileges
-  // when using FileTable ExternalStorage.
-  optional string user_proto = 6 [(gogoproto.nullable) = false, (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/security.SQLUsernameProto"];
-}
-
-// ParquetWriterSpec is the specification for a processor that consumes rows and
-// writes them to Parquet files at uri. It outputs a row per file written with
-// the file name, row count and byte size.
-message ParquetWriterSpec {
-  // destination as a cloud.ExternalStorage URI pointing to an export store
-  // location (directory).
-  optional string destination = 1 [(gogoproto.nullable) = false];
-  optional string name_pattern = 2 [(gogoproto.nullable) = false];
-  optional roachpb.ParquetOptions options = 3 [(gogoproto.nullable) = false];
+  optional roachpb.IOFileFormat format = 3 [(gogoproto.nullable) = false];
 
   // chunk_rows is num rows to write per file. 0 = no limit.
   optional int64 chunk_rows = 4 [(gogoproto.nullable) = false];
   // chunk_size is the target byte size per file.
-  optional int64 chunk_size = 7 [(gogoproto.nullable) = false];
-
-  // compression_codec specifies compression used for exported file. THIS IS ISN'T USED RIGHT NOW
-  optional FileCompression compression_codec = 5 [(gogoproto.nullable) = false];
+  optional int64 chunk_size = 5 [(gogoproto.nullable) = false];
 
   // User who initiated the export. This is used to check access privileges
   // when using FileTable ExternalStorage.
   optional string user_proto = 6 [(gogoproto.nullable) = false, (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/security.SQLUsernameProto"];
 
   // col_names specifies the logical column names for the exported parquet file.
-  repeated string col_names = 8 ;
-
-  // col_nullability specifies which columns allow null values in the exported parquet file.
-  repeated bool col_nullability = 9 ;
+  repeated string col_names = 7 ;
 }
 
 // BulkRowWriterSpec is the specification for a processor that consumes rows and

--- a/pkg/sql/export.go
+++ b/pkg/sql/export.go
@@ -20,7 +20,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/featureflag"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings"
-	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/opt/exec"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
@@ -41,13 +40,10 @@ type exportNode struct {
 	// fileNamePattern represents the file naming pattern for the
 	// export, typically to be appended to the destination URI
 	fileNamePattern string
-	csvOpts         *roachpb.CSVOptions
-	parquetOpts     *roachpb.ParquetOptions
+	format          roachpb.IOFileFormat
 	chunkRows       int
 	chunkSize       int64
-	fileCompression execinfrapb.FileCompression
 	colNames        []string
-	colNullability  []bool
 }
 
 func (e *exportNode) startExec(params runParams) error {
@@ -73,6 +69,15 @@ const (
 	exportOptionChunkSize   = "chunk_size"
 	exportOptionFileName    = "filename"
 	exportOptionCompression = "compression"
+
+	exportChunkSizeDefault = int64(32 << 20) // 32 MB
+	exportChunkRowsDefault = 100000
+
+	exportFilePatternPart = "%part%"
+	exportGzipCodec       = "gzip"
+	exportSnappyCodec     = "snappy"
+	csvSuffix             = "csv"
+	parquetSuffix         = "parquet"
 )
 
 var exportOptionExpectValues = map[string]KVStringOptValidate{
@@ -83,13 +88,6 @@ var exportOptionExpectValues = map[string]KVStringOptValidate{
 	exportOptionCompression: KVStringOptRequireValue,
 	exportOptionChunkSize:   KVStringOptRequireValue,
 }
-
-const exportChunkSizeDefault = int64(32 << 20) // 32 MB
-const exportChunkRowsDefault = 100000
-const exportFilePatternPart = "%part%"
-const exportCompressionCodec = "gzip"
-const csvSuffix = "csv"
-const parquetSuffix = "parquet"
 
 // featureExportEnabled is used to enable and disable the EXPORT feature.
 var featureExportEnabled = settings.RegisterBoolSetting(
@@ -103,11 +101,11 @@ var featureExportEnabled = settings.RegisterBoolSetting(
 func (ef *execFactory) ConstructExport(
 	input exec.Node,
 	fileName tree.TypedExpr,
-	fileFormat string,
+	fileSuffix string,
 	options []exec.KVOption,
 	notNullCols exec.NodeColumnOrdinalSet,
 ) (exec.Node, error) {
-	fileFormat = strings.ToLower(fileFormat)
+	fileSuffix = strings.ToLower(fileSuffix)
 
 	if !featureExportEnabled.Get(&ef.planner.ExecCfg().Settings.SV) {
 		return nil, pgerror.Newf(
@@ -129,8 +127,8 @@ func (ef *execFactory) ConstructExport(
 		return nil, errors.Errorf("EXPORT cannot be used inside a transaction")
 	}
 
-	if fileFormat != csvSuffix && fileFormat != parquetSuffix {
-		return nil, errors.Errorf("unsupported export format: %q", fileFormat)
+	if fileSuffix != csvSuffix && fileSuffix != parquetSuffix {
+		return nil, errors.Errorf("unsupported export format: %q", fileSuffix)
 	}
 
 	destinationDatum, err := fileName.Eval(ef.planner.EvalContext())
@@ -162,12 +160,6 @@ func (ef *execFactory) ConstructExport(
 		return nil, err
 	}
 
-	var (
-		exportFilePattern string
-		csvOpts           *roachpb.CSVOptions
-		parquetOpts       *roachpb.ParquetOptions
-	)
-
 	cols := planColumns(input.(planNode))
 	colNames := make([]string, len(cols))
 	colNullability := make([]bool, len(cols))
@@ -176,9 +168,10 @@ func (ef *execFactory) ConstructExport(
 		colNullability[i] = !notNullCols.Contains(i)
 	}
 
-	switch fileFormat {
+	format := roachpb.IOFileFormat{}
+	switch fileSuffix {
 	case csvSuffix:
-		csvOpts = &roachpb.CSVOptions{}
+		csvOpts := roachpb.CSVOptions{}
 		if override, ok := optVals[exportOptionDelimiter]; ok {
 			csvOpts.Comma, err = util.GetSingleRune(override)
 			if err != nil {
@@ -188,11 +181,14 @@ func (ef *execFactory) ConstructExport(
 		if override, ok := optVals[exportOptionNullAs]; ok {
 			csvOpts.NullEncoding = &override
 		}
-		exportFilePattern = exportFilePatternPart + "." + csvSuffix
-
+		format.Format = roachpb.IOFileFormat_CSV
+		format.Csv = csvOpts
 	case parquetSuffix:
-		parquetOpts = &roachpb.ParquetOptions{}
-		exportFilePattern = exportFilePatternPart + "." + parquetSuffix
+		parquetOpts := roachpb.ParquetOptions{
+			ColNullability: colNullability,
+		}
+		format.Format = roachpb.IOFileFormat_Parquet
+		format.Parquet = parquetOpts
 	}
 
 	chunkRows := exportChunkRowsDefault
@@ -219,28 +215,30 @@ func (ef *execFactory) ConstructExport(
 
 	// Check whenever compression is expected and extract compression codec name in case
 	// of positive result
-	var codec execinfrapb.FileCompression
+	var codec roachpb.IOFileFormat_Compression
 	if name, ok := optVals[exportOptionCompression]; ok && len(name) != 0 {
-		if strings.EqualFold(name, exportCompressionCodec) {
-			codec = execinfrapb.FileCompression_Gzip
-		} else {
+		switch {
+		case strings.EqualFold(name, exportGzipCodec):
+			codec = roachpb.IOFileFormat_Gzip
+		case strings.EqualFold(name, exportSnappyCodec) && fileSuffix == parquetSuffix:
+			codec = roachpb.IOFileFormat_Snappy
+		default:
 			return nil, pgerror.Newf(pgcode.InvalidParameterValue,
-				"unsupported compression codec %s", name)
+				"unsupported compression codec %s for %s file format", name, fileSuffix)
 		}
+		format.Compression = codec
 	}
 
 	exportID := ef.planner.stmt.QueryID.String()
+	exportFilePattern := exportFilePatternPart + "." + fileSuffix
 	namePattern := fmt.Sprintf("export%s-%s", exportID, exportFilePattern)
 	return &exportNode{
 		source:          input.(planNode),
 		destination:     string(*destination),
 		fileNamePattern: namePattern,
-		csvOpts:         csvOpts,
-		parquetOpts:     parquetOpts,
+		format:          format,
 		chunkRows:       chunkRows,
 		chunkSize:       chunkSize,
-		fileCompression: codec,
 		colNames:        colNames,
-		colNullability:  colNullability,
 	}, nil
 }

--- a/pkg/sql/rowexec/processors.go
+++ b/pkg/sql/rowexec/processors.go
@@ -13,6 +13,7 @@ package rowexec
 import (
 	"context"
 
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
@@ -274,25 +275,17 @@ func NewProcessor(
 		}
 		return NewStreamIngestionDataProcessor(flowCtx, processorID, *core.StreamIngestionData, post, outputs[0])
 	}
-	if core.CSVWriter != nil {
+	if core.Exporter != nil {
 		if err := checkNumInOut(inputs, outputs, 1, 1); err != nil {
 			return nil, err
 		}
-		if NewCSVWriterProcessor == nil {
-			return nil, errors.New("CSVWriter processor unimplemented")
+
+		if core.Exporter.Format.Format == roachpb.IOFileFormat_Parquet {
+			return NewParquetWriterProcessor(flowCtx, processorID, *core.Exporter, inputs[0], outputs[0])
 		}
-		return NewCSVWriterProcessor(flowCtx, processorID, *core.CSVWriter, inputs[0], outputs[0])
+		return NewCSVWriterProcessor(flowCtx, processorID, *core.Exporter, inputs[0], outputs[0])
 	}
-	if core.ParquetWriter != nil {
-		if err := checkNumInOut(inputs, outputs, 1, 1); err != nil {
-			return nil, err
-		}
-		if NewParquetWriterProcessor == nil {
-			return nil, errors.New("ParquetWriter processor unimplemented")
-		}
-		return NewParquetWriterProcessor(flowCtx, processorID, *core.ParquetWriter, inputs[0],
-			outputs[0])
-	}
+
 	if core.BulkRowWriter != nil {
 		if err := checkNumInOut(inputs, outputs, 1, 1); err != nil {
 			return nil, err
@@ -392,10 +385,10 @@ var NewRestoreDataProcessor func(*execinfra.FlowCtx, int32, execinfrapb.RestoreD
 var NewStreamIngestionDataProcessor func(*execinfra.FlowCtx, int32, execinfrapb.StreamIngestionDataSpec, *execinfrapb.PostProcessSpec, execinfra.RowReceiver) (execinfra.Processor, error)
 
 // NewCSVWriterProcessor is implemented in the non-free (CCL) codebase and then injected here via runtime initialization.
-var NewCSVWriterProcessor func(*execinfra.FlowCtx, int32, execinfrapb.CSVWriterSpec, execinfra.RowSource, execinfra.RowReceiver) (execinfra.Processor, error)
+var NewCSVWriterProcessor func(*execinfra.FlowCtx, int32, execinfrapb.ExportSpec, execinfra.RowSource, execinfra.RowReceiver) (execinfra.Processor, error)
 
 // NewParquetWriterProcessor is implemented in the non-free (CCL) codebase and then injected here via runtime initialization.
-var NewParquetWriterProcessor func(*execinfra.FlowCtx, int32, execinfrapb.ParquetWriterSpec, execinfra.RowSource, execinfra.RowReceiver) (execinfra.Processor, error)
+var NewParquetWriterProcessor func(*execinfra.FlowCtx, int32, execinfrapb.ExportSpec, execinfra.RowSource, execinfra.RowReceiver) (execinfra.Processor, error)
 
 // NewChangeAggregatorProcessor is implemented in the non-free (CCL) codebase and then injected here via runtime initialization.
 var NewChangeAggregatorProcessor func(*execinfra.FlowCtx, int32, execinfrapb.ChangeAggregatorSpec, *execinfrapb.PostProcessSpec, execinfra.RowReceiver) (execinfra.Processor, error)

